gcloud-java-core/src/main/java/com/google/gcloud/BaseServiceException.java

```diff
@@ -147,15 +162,15 @@ public BaseServiceException(int code, String message, String reason, boolean ide
     return Collections.emptySet();
   }
 
-  protected boolean isRetryable(GoogleJsonError error) {
-    return error != null && error(error).isRetryable(retryableErrors());
+  protected boolean isRetryable(boolean idempotent, GoogleJsonError error) {
```

Comment:
Instead of this method, which I think we should probably embed in the `isRetryable(boolean, IOException)` I think we should have `protected boolean isRetryable(boolean idempotent, Error error)`. default implementation would just delegated to `error.isRetryable(idempotent, retryableErrors())`.

Having that I would change the first constructor and the last constructor to call it instead.